File: 234-palindrome-linked-list/234-palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
      
        vals = [] 
        h =head 
        
        while h is not None:
            vals.append(h.val)
            h = h.next 
        return vals == vals[::-1]

File: 234-palindrome-linked-list/234-palindrome-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* tmp ; 
    bool isPalindrome(ListNode* head) {
        
        tmp = head ; 
        return helper(head); 
    }
    bool helper(ListNode* p){
        if(NULL == p) return true ; 
        bool pal = helper(p->next)&( tmp ->val == p->val) ; 
        tmp = tmp->next ; 
        return pal ; 
    }


        
        
    
};